Feature: Show group name as part of the default vehicle name.

stormcone opened the pull request:
Only if the vehicle is member of a group and does not have a user defined name.

This patch is based on @KeldorKatarn's commit:
KeldorKatarn/OpenTTD_PatchPack@3f03ec47

![group_name](https://user-images.githubusercontent.com/48624099/93125693-cb6e0300-f6cb-11ea-9d98-73ac06c5e482.png)

stormcone committed:
Feature: Show group name as part of the default vehicle name.

Only if the vehicle is member of a group and does not have a user defined name.

TrueBrain commented:
I really like this. There is one thing my OCD triggers on ... but I am not sure it needs fixing:

It is now called `Train 1`  but `MyGroup #1`. Depending on if it is in a group, the number gained a `#`.

I am almost tempted to always add a `#` :P

TrueBrain committed:
Fix: prefix the Nth vehicle with "#" when ungrouped

This is for consistency; most other objects add a # to indicate
it is the Nth of that object, except for Trains / RVs / Ships /
Aircrafts.
This becomes painfully apparent with unnamed vehicles in groups,
which do get a "#".

TrueBrain commented:
Okay, this hit my OCD enough that I added a commit to change `Train 1` into `Train #1`. Will be a while before all translations pick that up.

I made some minor changes to your patch:
- With `AddToGroup`, only invalidate the data of the vehicle that is added
- With `DeleteGroup`, no need to invalidate data; the calls done by `DeleteGroup` always cause `AddToGroup` to be called
- Renamed the new string `STR_FORMAT_GROUP_VEHICLE_NAME` to include `_NAME`

Also rebased :)

stormcone commented:
@TrueBrain Okay, thank you. :)
